.github/workflows/_rust.yml

```diff
@@ -143,41 +146,36 @@ jobs:
-          cp "../target/release/firezone_gui_client.pdb" "${{ env.BINARY_DEST_PATH }}-x64.pdb"
-
-          Get-FileHash ${{ env.BINARY_DEST_PATH }}-x64.exe -Algorithm SHA256 | Select-Object Hash > ${{ env.BINARY_DEST_PATH }}-x64.exe.sha256sum.txt
-          Get-FileHash ${{ env.BINARY_DEST_PATH }}-x64.msi -Algorithm SHA256 | Select-Object Hash > ${{ env.BINARY_DEST_PATH }}-x64.msi.sha256sum.txt
-          Get-FileHash ${{ env.BINARY_DEST_PATH }}-x64.pdb -Algorithm SHA256 | Select-Object Hash > ${{ env.BINARY_DEST_PATH }}-x64.pdb.sha256sum.txt
-
-          # This might catch regressions in #3384, depending how CI runners
-          # handle exit codes
-          git diff --exit-code
-      - name: Save GUI client
-        uses: actions/upload-artifact@v4
+      - name: Rename artifacts and compute SHA256
+        shell: bash
+        run: ${{ matrix.rename-script }}
+      - name: Save Linux client
+        if: ${{ runner.os == 'Linux' }}
+        uses: ./.github/actions/upload-gui-artifact
+        with:
+          base: ${{ env.BINARY_DEST_PATH }}-amd64
+      - name: Save Linux AppImage
+        if: ${{ runner.os == 'Linux' }}
+        uses: ./.github/actions/upload-gui-artifact
+        with:
+          base: ${{ env.BINARY_DEST_PATH }}_amd64.AppImage
+      - name: Save Linux deb package
+        if: ${{ runner.os == 'Linux' }}
+        uses: ./.github/actions/upload-gui-artifact
+        with:
+          base: ${{ env.BINARY_DEST_PATH }}_amd64.deb
+      - name: Save Windows client
+        if: ${{ runner.os == 'Windows' }}
+        uses: ./.github/actions/upload-gui-artifact
         with:
-          name: ${{ env.BINARY_DEST_PATH }}-x64
-          path: |
-            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}-x64.exe
-            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}-x64.exe.sha256sum.txt
+          base: ${{ env.BINARY_DEST_PATH }}-x64.exe
       - name: Save Windows MSI installer
         if: ${{ runner.os == 'Windows' }}
-        uses: actions/upload-artifact@v4
+        uses: ./.github/actions/upload-gui-artifact
         with:
-          name: ${{ env.BINARY_DEST_PATH }}-x64-msi
-          path: |
-            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}-x64.msi
-            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}-x64.msi.sha256sum.txt
+          base: ${{ env.BINARY_DEST_PATH }}-x64.msi
       - name: Save Windows debug symbols
         if: ${{ runner.os == 'Windows' }}
-        uses: actions/upload-artifact@v4
+        uses: ./.github/actions/upload-gui-artifact
         with:
-          name: ${{ env.BINARY_DEST_PATH }}-x64-pdb
-          path: |
-            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}-x64.pdb
-            ${{ github.workspace }}/rust/gui-client/${{ env.BINARY_DEST_PATH }}-x64.pdb.sha256sum.txt
+          base: ${{ env.BINARY_DEST_PATH }}-x64.pdb
```

Comment:
Could this all be simplified to just

```yaml
     matrix:
        include:
          - runs-on: ubuntu-20.04
            binary-dest-path: firezone-linux-gui-client
            rename-script: ../../scripts/build/tauri-rename-ubuntu.sh
            artifacts:
              - rust/gui-client/firezone-linux-gui-client-amd64
              - rust/gui-client/firezone-linux-gui-client-amd64.sha256sum.txt
              - rust/gui-client/firezone-linux-gui-client-amd64.AppImage
              - rust/gui-client/firezone-linux-gui-client-amd64.AppImage.sha256sum.txt
              - rust/gui-client/firezone-linux-gui-client-amd64.deb
              - rust/gui-client/firezone-linux-gui-client-amd64.deb.sha256sum.txt
          - runs-on: windows-2019
            binary-dest-path: firezone-windows-client
            rename-script: ../../scripts/build/tauri-rename-windows.sh
            artifacts:
              - rust/gui-client/firezone-windows-client-x64.msi
              - rust/gui-client/firezone-windows-client-x64.msi.sha256sum.txt
              - rust/gui-client/firezone-windows-client-x64.exe
              - rust/gui-client/firezone-windows-client-x64.exe.sha256sum.txt
              - rust/gui-client/firezone-windows-client-x64.pdb
              - rust/gui-client/firezone-windows-client-x64.pdb.sha256sum.txt
    - uses: actions/upload-artifact@v4
      with:
        name: ${{ matrix.binary-dest-path }}
        path: ${{ matrix.artifacts }}

```

Reply:
It could, but then it'll produce one big zip file with the AppImage and deb and exe.

Like it's not wrong, I'd just be downloading 67 MB of WebView and then unpacking it somewhere every time I wanted to check the CI output.